src/api/public/notes/notes.controller.ts

```diff
@@ -1,53 +1,98 @@
 import {
+  BadRequestException,
   Body,
   Controller,
   Delete,
   Get,
   Header,
+  Logger,
   Param,
   Post,
   Put,
+  Req,
 } from '@nestjs/common';
+import { Request } from 'express';
+import * as getRawBody from 'raw-body';
 import { NotePermissionsUpdateDto } from '../../../notes/note-permissions.dto';
 import { NotesService } from '../../../notes/notes.service';
 import { RevisionsService } from '../../../revisions/revisions.service';
 
 @Controller('notes')
 export class NotesController {
+  private readonly logger = new Logger(NotesController.name);
+
   constructor(
     private noteService: NotesService,
     private revisionsService: RevisionsService,
   ) {}
 
+  /**
+   * Extract the raw markdown from the request body and create a new note with it
+   *
+   * Implementation inspired by https://stackoverflow.com/questions/52283713/how-do-i-pass-plain-text-as-my-request-body-using-nestjs
+   */
   @Post()
-  createNote(@Body() noteContent: string) {
-    return this.noteService.createNote(noteContent);
+  async createNote(@Req() req: Request) {
+    // we have to check req.readable because of raw-body issue #57
+    // https://github.com/stream-utils/raw-body/issues/57
+    if (req.readable) {
+      let bodyText: string = await getRawBody(req, 'utf-8');
+      bodyText = bodyText.trim();
+      this.logger.debug('Got raw markdown:\n' + bodyText);
+      return this.noteService.createNoteDto(bodyText);
+    } else {
+      // TODO: Better error message
+      throw new BadRequestException('Invalid body');
```

Comment:
maybe this should call `createNamedNote` or at least call a common function so the code needs only to exist once?

Reply:
If you refer to the duplicated code for extracting the request body, I openend https://github.com/codimd/server/issues/501 to move that into a decorator.